lib/element.ts

```diff
@@ -1166,6 +1173,34 @@ export class ElementFinder extends WebdriverWebElement {
         (element as any).getWebElement ? (element as ElementFinder).getWebElement() :
                                          element as WebElement);
   }
+
+  /**
+   * Get the visible innerText of this element, including sub-elements, without
+   * any leading or trailing whitespace. Visible elements are not hidden by CSS.
+   * Works also for input fields
+   *
+   * @view
+   * <div id="foo" class="bar">Inner text</div>
+   *
+   * @example
+   * var foo = element(by.id('foo'));
+   * expect(foo.getText()).toEqual('Inner text');
+   *
+   * @returns {!webdriver.promise.Promise.<string>} A promise that will be
+   *     resolved with the element's visible text.
+   */
+  getText(): wdpromise.Promise<string> {
+    let webElem = this.getWebElement();
+    return webElem.getText().then((text) => {
+      if (text) {
+        return text.trim();
```

Comment:
yes I thought that this is a little dangerous. I created this here because of IE or Edge who sometimes appends whitespaces on the text (my previous commit failed because of this).

Reply:
yes, I think we should try not to change the original behavior.
For now, we can "correct" our smoke test by trimming the actual result in test, and create an issue/bug to change it back after Microsoft fixed that bug.